lib/util/url.ts

```diff
@@ -79,3 +81,18 @@ export function parseUrl(url: string): URL | null {
 export function createURLFromHostOrURL(url: string): URL | null {
   return parseUrl(url) ?? parseUrl(`https://${url}`);
 }
+
+let parseLinkHeaderPromise: Promise<typeof _parseLinkHeader> | null = null;
+
+export type LinkHeaderLinks = _parseLinkHeader.Links;
+
+export async function parseLinkHeader(
+  linkHeader: string | null | undefined
+): Promise<LinkHeaderLinks | null> {
+  if (null === parseLinkHeaderPromise) {
+    // https://github.com/thlorenz/parse-link-header#environmental-variables
+    process.env.PARSE_LINK_HEADER_THROW_ON_MAXLEN_EXCEEDED = 'true';
+    parseLinkHeaderPromise = import('parse-link-header').then((m) => m.default);
+  }
+  return (await parseLinkHeaderPromise)(linkHeader);
```

Comment:
Should we catch and log errors here?

Reply:
this is getting pretty ugly now..

Do we think there's much chance of headers over 2k that we want?

Reply:
I think normally not. That header shouldn't be that long. I'll simplify this PR to use normal import.